pkg/plugin/settings.go

```diff
@@ -39,9 +40,79 @@ func (settings *Settings) isValid() (err error) {
 
 // LoadSettings will read and validate Settings from the DataSourceConfig
 func LoadSettings(config backend.DataSourceInstanceSettings) (settings Settings, err error) {
-	if err := json.Unmarshal(config.JSONData, &settings); err != nil {
+	var jsonData map[string]interface{}
+	if err := json.Unmarshal(config.JSONData, &jsonData); err != nil {
 		return settings, fmt.Errorf("%s: %w", err.Error(), ErrorMessageInvalidJSON)
 	}
+
+	if jsonData["server"] != nil {
+		settings.Server = jsonData["server"].(string)
+	}
+	if jsonData["port"] != nil {
+		if port, err := strconv.ParseInt(jsonData["port"].(string), 0, 64); err == nil {
```

Comment:
Is it always a string? Do we want to account for strings _and_ numbers?

Reply:
Literally my latest commit 😄